rero_ils/modules/items/api_views.py

```diff
@@ -346,56 +346,63 @@ def item_availability(item_pid):
     })
 
 
-@api_blueprint.route(
-    '/<item_pid>/can_request/<library_pid>/<patron_barcode>', methods=['GET'])
+@api_blueprint.route('/<item_pid>/can_request', methods=['GET'])
 @check_authentication
 @jsonify_error
-def can_request(item_pid, library_pid, patron_barcode):
-    """HTTP request to check if a librarian can request an item for a patron.
+def can_request(item_pid):
+    """HTTP request to check if an item can be requested.
 
-    required_parameters: item_pid, library_pid, patron_barcode
-    """
-    return is_librarian_can_request_item_for_patron(
-        item_pid, library_pid, patron_barcode)
+    Depending of query string argument, either only check if configuration
+    allows the request of this item ; either if a librarian can request an
+    item for a patron.
 
+    `api/item/<item_pid>/can_request` :
+         --> only check config
+    `api/item/<item_pid>/can_request?library=<library_pid>&patron=<barcode>`:
+         --> check if the patron can request this item (check the cipo)
+    """
+    item = Item.get_record_by_pid(item_pid)
+    library_pid = flask_request.args.get('library')
+    patron_barcode = flask_request.args.get('patron')
```

Comment:
if you are sending the patron barcode, I would name it only patron_barcode instead of patron. because now you think it is the  patron record

Reply:
done